pynamodb/models.py

```diff
@@ -232,12 +237,16 @@ class Model(AttributeContainer):
     _index_classes = None
     DoesNotExist = DoesNotExist
 
-    def __init__(self, hash_key=None, range_key=None, **attributes):
+    def __init__(self, hash_key=None, range_key=None, previously_saved=False, **attributes):
```

Comment:
Maybe use a name that has a lower chance of being used by existing code, e.g. `__previously_saved`?

Reply:
that makes it a class var and it makes tests fail.. changing back

Reply:
A ... class var?!

Reply:
I was too vague.. https://stackoverflow.com/questions/1301346/what-is-the-meaning-of-a-single-and-a-double-underscore-before-an-object-name

Reply:
```py
class Foo:
    def __init__(self, abc, __xyz):
        self.abc = abc
        self.xyz = __xyz

foo = Foo('abc', 'xyz')
print(foo.abc)
print(foo.xyz)
```
prints:
```
abc
xyz
```

Reply:
yea but I removed the self.xyz part.  I don't think 2 underscores are necessary.. I don't think there will be collisions.  It's just a parameter like `is_new_object`

Reply:
The point is, if someone currently has:
```
class MyModel(Model):
   is_new_object = UnicodeAttribute()
   ...
```
and does:
```
my_model = MyModel(is_new_object="hello world")
```
then before your change, their code is valid, and after your change, it has unexpected side effects. That's why I'm recommending a parameter name that would have lower chances of colliding with an existing attribute, especially since we're only using it as a means of internal signaling.